persistence/migrations/4_add_status.up.sql

```diff
@@ -0,0 +1,34 @@
+CREATE TYPE invoice_status as ENUM (
+    'SETTLE_REQUESTED',
+    'SETTLED',
+    'FAILED'
+);
+
+-- Migrate invoices table
+ALTER TABLE lnmux.invoices ADD "status" invoice_status;
+ALTER TABLE lnmux.invoices ADD "finalized_at" TIMESTAMPTZ;
+
+UPDATE lnmux.invoices SET
+        status = CASE WHEN settled = FALSE THEN 'SETTLE_REQUESTED'::invoice_status ELSE 'SETTLED'::invoice_status END,
+        finalized_at = CASE WHEN settled = TRUE THEN settled_at ELSE NULL END;
+
+ALTER TABLE lnmux.invoices DROP COLUMN "settled";
+ALTER TABLE lnmux.invoices DROP COLUMN "settled_at";
```

Comment:
I think that postgres also has a RENAME COLUMN statement. But this works too.

Reply:
Indeed, I didn't know it exists. It could be convenient for the migrate_up, but for the migrate_down, we still need to use the DROP (or a more complex logic) because for invoice with 'FAILED' status, `finalized_at` will be not NULL but we don't want `settled_at` to be not null (I think) 

I think I will keep the drop for now (if really no objections)